test/parallel/test-domain-timers-uncaught-exception.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+const common = require('../common');
+
+// This test ensures that the timer callbacks are
+// called in the order in which they were created
+// in the event of an unhandled exception in the domain
+
+const domain = require('domain').create();
+const assert = require('assert');
+
+var first = false;
+
+domain.run(function() {
+  setTimeout(function() { throw Error('FAIL'); }, 1);
+  setTimeout(function() { first = true; }, 1);
+  setTimeout(function() { assert.strictEqual(first, true);  }, 2);
+});
+
+domain.once('error', common.mustCall(() => {}));
```

Comment:
I assume the `err` arg here must be that `'FAIL'` one, but you could make it explicit